src/coreclr/jit/optimizebools.cpp

```diff
@@ -1570,3 +1570,1008 @@ PhaseStatus Compiler::optOptimizeBools()
+//
+// Arguments:
+//    patternVal - the pattern value to set
+//    node - the node that contains the patternVal
+//
+void OptRangePatternDsc::optSetMinPattern(ssize_t patternVal, GenTree* node)
+{
+    if (patternVal < m_minPattern)
+    {
+        m_minPattern = patternVal;
+
+        // Update minOp to the tree with the min pattern
+        optSetMinOp(node);
+    }
+}
+
+void OptRangePatternDsc::optSetMaxPattern(ssize_t patternVal)
+{
+    if (patternVal > m_maxPattern)
+    {
+        m_maxPattern = patternVal;
+    }
+}
+
+void OptRangePatternDsc::optSetMinOp(GenTree* minOpNode)
+{
+    assert(minOpNode != nullptr && minOpNode->IsIntegralConst());
+    m_minOp = minOpNode;
+}
+
+//-----------------------------------------------------------------------------
+// optGetJumpTargetBB:  Get jumpTargetBB for the pattern
+//
+// Arguments:
+//    block - the basic block to get its jump target.
+//
+// Return Value:
+//    The jump target BB for the pattern
+//
+// Notes:
+//   If compare operator is GT_EQ, the switch jump target (true case) is bbJumpDest.
+//   If compare operator is GT_NE,it is bbNext.
+//
+BasicBlock* OptRangePatternDsc::optGetJumpTargetBB(BasicBlock* block)
+{
+    assert(block != nullptr && (block->bbJumpKind == BBJ_SWITCH || block->bbJumpKind == BBJ_COND));
+    assert(block->lastStmt()->GetRootNode()->gtGetOp1() != nullptr);
+
+    GenTree* op1  = block->lastStmt()->GetRootNode()->gtGetOp1();
+    auto     oper = op1->OperGet();
+    assert(oper == GT_EQ || oper == GT_NE);
+
+    if (oper == GT_EQ)
+    {
+        return block->bbJumpDest;
+    }
+    else if (oper == GT_NE)
+    {
+        return block->bbNext;
+    }
+
+    return nullptr;
+}
+
+//-----------------------------------------------------------------------------
+// optSetDefaultJmpBB:  Get Switch Default jump target
+//
+// Arguments:
+//    defaultJmpBB - the basic block to set its jump target.
+//
+// Return Value:
+//    true if the m_defaultJmpBB is set, false otherwise.
+//
+// Notes:
+//   For GT_EQ, Swithc Default jump target (for false case) is set to bbNext.
+//   For GT_NE, Swithc Default jump target (for false case) is set to bbJumpDest.
+//
+void OptRangePatternDsc::optSetDefaultJmpBB(BasicBlock* block)
+{
+    assert(block != nullptr && block->bbJumpKind == BBJ_COND);
+    assert(block->lastStmt()->GetRootNode()->gtGetOp1() != nullptr);
+
+    auto oper = block->lastStmt()->GetRootNode()->gtGetOp1()->OperGet();
+    assert(oper == GT_EQ || oper == GT_NE);
+
+    if (oper == GT_EQ)
+    {
+        m_defaultJmpBB = block->bbNext;
+    }
+    else if (oper == GT_NE)
+    {
+        m_defaultJmpBB = block->bbJumpDest;
+    }
+}
+
+//-----------------------------------------------------------------------------
+// optGetDefaultJmpBB:  Get Switch Default jump target
+//
+// Return Value:
+//    The default jump target BB of SWITCH block
+//
+BasicBlock* OptRangePatternDsc::optGetDefaultJmpBB()
+{
+    return m_defaultJmpBB;
```

Comment:
it feels like your class has many very simple methods/accessors which bloat the code size - maybe consider replacing simple ones like this with just accessing `m_defaultJmpBB` directly?